### PR TITLE
Address security and linting for passport-azure-ad

### LINKED ISSUE #3914: Old version of https-proxy-agent
Hi,

Could you please take a look at upgrading the version of https-proxy-agent in your dependencies, you are currently on 2.4.0 which uses an old version of agent-base (<6.0.0) and this is causing issues with a package that depends on you.
https://github.com/joeyboey/node-red-contrib-passport-azure-ad/issues/1

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR adds linting and updates dependencies for the passport-azure-ad library. 

- Upgrades `grunt-contrib-nodeunit`
- Upgrades `https-proxy-agent`, addressing issue #3914 
- Upgrades `lodash` in the package-lock, addressing security advisory [1673](https://www.npmjs.com/advisories/1673)
- Upgrades `path-parse` in the package-lock, addressing security advisory [1773](https://www.npmjs.com/advisories/1773)
- Upgrades `diff` in the package-lock, addressing security advisory [1631](https://www.npmjs.com/advisories/1631)
- Adds `eslint`, `eslint-plugin-header`, `eslint-plugin-security` and lint rules in line with the rest of MSAL.js libraries. Lint errors and warnings addressed.

### CODE REVIEW

File: maintenance/passport-azure-ad/lib/aadutils.js [hunk 1]
```diff
@@ -1,36 +1,21 @@
...
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
  */
-'use strict';
 
-const base64url = require('base64url');
-const crypto = require('crypto');
-const util = require('util');
+/* eslint-disable security/detect-new-buffer */
```
Comment (peterzenz) [inline maintenance/passport-azure-ad/lib/aadutils.js:6]: If we're disabling checks here, we should explicitly document why it's being disabled and the scope it's disabled in should be no larger than what's necessary.

File: maintenance/passport-azure-ad/lib/aadutils.js [hunk 2]
```diff
@@ -253,66 +246,80 @@ exports.getErrorMessage = (err) => {
...
-// should be backward compatible as RFC 6265 specifies that browsers should
-// ignore unknown cookie attributes. However, for the specific case of the
-// SameSite attribute, some browsers incorrectly implemented the attribute or
-//  implemented an earlier draft which had contradictory behavior.
-// For these browsers which attempted to support SameSite,
-// but have bugs in their support, we want to omit the SameSite=None attribute.
-// See Chromium official guidance here:  https://www.chromium.org/updates/same-site/incompatible-clients
+/*
+ * This is a list maintained by the AAD server team, will need to keep an eye on this
+ * as things change.  Not ideal, but it is what it is
```
Comment (peterzenz) [inline maintenance/passport-azure-ad/lib/aadutils.js:266]: "Not ideal, but it is what it is"

I realize text is is the older iteration, but can we remove opinions on implementations here?

File: maintenance/passport-azure-ad/lib/aadutils.js [hunk 3]
```diff
@@ -253,66 +246,80 @@ exports.getErrorMessage = (err) => {
...
+ * in general, a change like this (adding a new cookie attribute)
+ * should be backward compatible as RFC 6265 specifies that browsers should
+ * ignore unknown cookie attributes. However, for the specific case of the
+ * SameSite attribute, some browsers incorrectly implemented the attribute or
+ *  implemented an earlier draft which had contradictory behavior.
```
Comment (peterzenz) [inline maintenance/passport-azure-ad/lib/aadutils.js:271]: Nit:  *  implemented...
Bonus space after the *

File: maintenance/passport-azure-ad/lib/jsonWebToken.js [hunk 4]
```diff
@@ -53,176 +36,200 @@ var hasCommonElem = (array1, array2) => {
  */
 exports.verify = function(jwtString, PEMKey, options, callback) {
 
-  /*********************************************************************
+  /**
```
Comment (peterzenz) [inline maintenance/passport-azure-ad/lib/jsonWebToken.js:39]: Formatting for these breaks is incorrect

File: maintenance/passport-azure-ad/lib/jsonWebToken.js [hunk 5]
```diff
@@ -53,176 +36,200 @@ var hasCommonElem = (array1, array2) => {
...
-  if (!Array.isArray(options.algorithms) || options.algorithms.length == 0 ||
-    (options.algorithms.length === 1 && options.algorithms[0] === 'none'))
-    return done(new Error('options.algorithms must be an array containing at least one algorithm'));
+    return done(new Error("options.algorithms is missing in jsonWebToken.verify"));
+  if (!Array.isArray(options.algorithms) || options.algorithms.length === 0 ||
+    (options.algorithms.length === 1 && options.algorithms[0] === "none"))
+    return done(new Error("options.algorithms must be an array containing at least one algorithm"));
 
-  /*********************************************************************
+  /**
```
Comment (peterzenz) [inline maintenance/passport-azure-ad/lib/jsonWebToken.js:83]: Formatting for these breaks is incorrect.  There are several more of these below in this file.

**Comment by jo-arroyo** [comment]
Closing due to failing tests due to upgrading mocha. Unable to revert. #3996 is replacement PR with same changes.

File: maintenance/passport-azure-ad/lib/aadutils.js (hunk 1 above)
Comment (jo-arroyo) [reply maintenance/passport-azure-ad/lib/aadutils.js:6]: Thanks, Peter. Reasons for disabled checks documented in new PR #3996

File: maintenance/passport-azure-ad/lib/aadutils.js (hunk 2 above)
Comment (jo-arroyo) [reply maintenance/passport-azure-ad/lib/aadutils.js:266]: Noted. Opinions removed on PR #3996

File: maintenance/passport-azure-ad/lib/aadutils.js (hunk 3 above)
Comment (jo-arroyo) [reply maintenance/passport-azure-ad/lib/aadutils.js:271]: Thanks. Changed on #3996

File: maintenance/passport-azure-ad/lib/jsonWebToken.js (hunk 4 above)
Comment (jo-arroyo) [reply maintenance/passport-azure-ad/lib/jsonWebToken.js:39]: Incorrect formatting removed on #3996

**Pull request closed without merging**